### DISCUSSION TITLE
How to set NodeSelector using jaeger operator

### DISCUSSION THREAD

**Discussion by ErikXu** [jaegertracing/jaeger-operator#1695] [Q&A]
I am using jaeger operator v1.22.0 to create jaeger instance. 
I want to set `NodeSelector` for modules like `collector`,  `query` and `agent`.
But I haven't found out the place after reading these two links:

https://www.jaegertracing.io/docs/1.22/operator/#updating-a-jaeger-instance-experimental

https://github.com/jaegertracing/jaeger-operator/blob/v1.22.0/pkg/apis/jaegertracing/v1/jaeger_types.go#L129

Please help to provide some examples.
Thanks!

**Comment by jpkrohling** [comment]
The suggested way is to use affinity/tolerations instead. Is there a reason to use node selectors instead?

**Reply by ErikXu** [reply]
Thanks for replying. After reading the doc of kubernetes again, I found I had a wrong concept of affinity before. It can really use affinity instead of node selectors in my scenarios.
